src/robusta/core/model/base_params.py

```diff
@@ -48,8 +60,106 @@ class GrafanaAnnotationParams(GrafanaParams):
     :var cluster_name: writen as one of the annotation's tags
     :var custom_tags: custom tags to add to the annotation
     """
+
     grafana_dashboard_panel: str = None
     cluster_name: str = None
     cluster_zone: str = None
     custom_tags: List[str] = None
 
+
+class PythonProcessParams(ActionParams):
+    """
+    :var process_substring: process name (or substring).
+    :var pid: pid
+    :var interactive: if more than one process matches, interactively ask which process to choose.
+    """
+
+    process_substring: str = ""
+    pid: int = None
+    interactive: bool = True
+
+    def has_exactly_one_match(self, all_processes: List[Process]) -> bool:
```

Comment:
I think the logic shouldn't be part of the base params.
We mix between the model and logic.
It doesn't seem like the right place logically.
In addition, it imports a lot of irrelevant stuff into base params which is used everywhere.
It will be harder to separate the params to a different module if we want it.
I would put that logic in a common utility class